tests/integration/components/sl-button-test.js

```diff
@@ -0,0 +1,206 @@
+    this.render( hbs`
+        {{#sl-button disabled=true}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.strictEqual(
+        this.$( '>:first-child' ).is( ':disabled' ),
+        true,
+        'Component becomes disabled'
+    );
+});
+
+test( 'Expected default classes are applied', function( assert ) {
+    this.render( template );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'btn' ),
+        'Has class "btn"'
+    );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'sl-button' ),
+        'Has class "sl-button"'
+    );
+});
+
+test( 'Labels are correctly initialized', function( assert ) {
+    this.render( hbs`
+        {{#sl-button label="Test"}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.equal(
+        this.$( '>:first-child' ).text().trim(),
+        'Test',
+        'Expected label is present as text'
+    );
+});
+
+test( 'Correct size class is set', function( assert ) {
+    this.render( hbs`
+        {{#sl-button size="large"}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'btn-lg' ),
+        'Has expected class "btn-lg"'
+    );
+
+    this.render( hbs`
+        {{#sl-button size="small"}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'btn-sm' ),
+        'Has expected class "btn-sm"'
+    );
+
+    this.render( hbs`
+        {{#sl-button size="extra-small"}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'btn-xs' ),
+        'Has expected class "btn-xs"'
+    );
+});
+
+test( 'Theme class is set correctly', function( assert ) {
+    this.render( hbs`
+        {{#sl-button theme="success"}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'btn-success' ),
+        'Has expected class "btn-success"'
+    );
+});
+
+test( 'Label changes for pending state', function( assert ) {
+    const pendingLabelText = 'Pending';
+    const staticText = 'Static';
+
+    this.set( 'staticText', staticText );
+
+    this.render( hbs`
+        {{#sl-button}}
+            {{staticText}}
+        {{/sl-button}}
+    ` );
+
+    assert.equal(
+        this.$( '>:first-child' ).text().trim(),
+        staticText,
+        'Static text is set initially'
+    );
+
+    this.set( 'pendingLabel', pendingLabelText );
+
+    this.render( hbs`
+        {{#sl-button pendingLabel=pendingLabel pending=true}}
+            Default Text
+            {{staticText}}
+        {{/sl-button}}
+    ` );
+
+    assert.equal(
+        this.$( '>:first-child' ).text().trim(),
+        pendingLabelText,
+        'Pending text is set while pending'
+    );
+});
+
+test( 'showModalWithStreamName property triggers modal to open', function( assert ) {
```

Comment:
This test should be broken up between Unit and Integration (with likely rename of the test(s)).

The unit test should assert that `send()` is called on the service and that `sendAction()` is called.

The integration test should assert that the `action` property is wired up correctly.